codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -102,6 +102,13 @@
  * implement all abstract methods properly.
  */
 public abstract class HttpObjectDecoder extends ByteToMessageDecoder {
+    public static final int DEFAULT_MAX_INITIAL_LINE_LENGTH = 4096;
```

Comment:
keep these fields pkg-private to reduce public API surface area?

Reply:
I would recommend making them public (my reason being the 2nd point in my motivation).

If I am a user of Netty today, and I want to instantiate an `HttpObjectDecoder` using default values, except I want to change only one parameter, e.g., `maxHeaderSize`, I am forced to give values for 3 other parameters as well. This requires users to manually declare values for those parameters, and they will typically need to inspect the defaults and repeat them.

Short of offering a more robust builder, exposing the defaults like this allows users to reference them accordingly and to only have to explicitly change the values they care about.

Reply:
I think expose these as public is ok.